### PR TITLE
Use `multipart_suggestions` more

### PR DESCRIPTION
**Pull request by estebank**
Built on top of #86532

### CODE REVIEW

**Comment by varkor** [comment]
As much as I'd like to, I'm not going to have time to review a PR of this size for a while (I'm not sure how much of that diff is just error output changes, though, which isn't so bad). If anyone else has time, they're welcome to take this over :)

**Comment by estebank** [comment]
@varkor no problem! Most of the diff is caused by the parent PR, until that is merged I would ignore this one.

**Comment by estebank** [comment]
@varkor I split out the part that is relevant and it should be relatively quick to review now.

File: compiler/rustc_typeck/src/check/demand.rs [hunk 1]
```diff
@@ -539,13 +543,14 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                 //                                   |     |
                                 //    consider dereferencing here: `*opt`  |
                                 // expected mutable reference, found enum `Option`
-                                if let Ok(src) = sm.span_to_snippet(left_expr.span) {
+                                if let Ok(_) = sm.span_to_snippet(left_expr.span) {
```
Comment (varkor) [inline compiler/rustc_typeck/src/check/demand.rs:546]: Can use `.is_ok()` here instead.

File: compiler/rustc_typeck/src/check/demand.rs [hunk 2]
```diff
@@ -584,24 +591,26 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     if let Some(call_span) =
                         iter::successors(Some(expr.span), |s| s.parent()).find(|&s| sp.contains(s))
                     {
-                        if let Ok(code) = sm.span_to_snippet(call_span) {
+                        if let Ok(_) = sm.span_to_snippet(call_span) {
```
Comment (varkor) [inline compiler/rustc_typeck/src/check/demand.rs:594]: Same here.

File: compiler/rustc_typeck/src/check/demand.rs [hunk 3]
```diff
@@ -584,24 +591,26 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
...
                                 Applicability::MachineApplicable,
+                                true,
                             ));
                         }
                     }
                     return None;
                 }
                 if sp.contains(expr.span) {
-                    if let Ok(code) = sm.span_to_snippet(expr.span) {
+                    if let Ok(_) = sm.span_to_snippet(expr.span) {
```
Comment (varkor) [inline compiler/rustc_typeck/src/check/demand.rs:607]: And here.

**Review by varkor — APPROVED** [review]
r=me after fixing the nits and squashing the last (rebase fixup) commit into the previous commit.

**Pull request merged**